### DISCUSSION TITLE
Q: How to query parameter of filament in a macro?

### DISCUSSION THREAD

**Discussion by cs301cs301** [Donkie/Spoolman#549] [Q&A]
Is there a way to query the filament parameters from the loaded spool in a Klipper macro?
e.g.
```
[gcode_macro SHOW_SPOOL_PARAMETER]
gcode:
    M118 {("Pressure Advance: ”, spool_id.filament.pressure_advance)}
```
Then you could use it to automatically adjust the retract, PA, etc. depending on the filament.

**Comment by bofh69** [comment]
A different way to solve that is to configure your slicer from the data in spoolman.
My program https://github.com/bofh69/spoolman2slicer does that for superslicer and OrcaSlicer. It should be easy to configure it for other slicers too.

You run it and it creates a file per filament from template files and the spoolman data.
